pkg/registration/register.go

```diff
@@ -1,18 +1,73 @@
 package registration
 
 import (
-	"github.com/SUSE/connect-ng/internal/collectors"
+	"encoding/json"
+
 	"github.com/SUSE/connect-ng/pkg/connection"
 )
 
+type announceRequest struct {
+	Hostname          string `json:"hostname"`
+	SystemInformation any    `json:"hwinfo"`
```

Comment:
I know the appeal with `any`, but I remember it being troublesome down the line when working on the branch that you pushed as the first PR from this series. If we expect this to be the `collectors` package, let's not shy away this fact. For now we can pull this internal library, but I'd expect to move the `collectors` package on the public library as well.

Reply:
I rather vote we define `collectors.Result` here as `registration.SystemInformation` or similar. So it can be populated.

Reply:
Gotcha, let's do that. Even though I believe that `collectors.Result` should be made public, as well as the rest of the `collectors` package. But this is a wildly new thing so don't worry about it for this PR.

Reply:
done